src/processor.js

```diff
@@ -128,8 +131,12 @@ class Processor {
           if (typeof node[name] === 'object' &&
             typeof nodeFilter[name] === 'object') {
             filterNode(node[name], nodeFilter[name]);
-            if (Object.keys(node[name]).length < 1) {
-              delete node[name];
+            try{
+              if (node && node[name] && Object.keys(node[name]).length < 1) {
+                delete node[name];
+              }
+            }catch(e){
+              console.log(e);
```

Comment:
I think we can avoid this catch that fails silently (just console message), if we check that `node && typeof node[name] === 'object'`, then there's no way this throws an error, eg:
``` js
             if (node && node[name] && typeof node[name] === 'object' && Object.keys(node[name]).length < 1) {
                delete node[name];
             }
```